internal/cli/opsmanager/clusters/delete.go

```diff
@@ -43,39 +48,162 @@ func (opts *DeleteOpts) Run() error {
 	if !opts.Confirm {
 		return nil
 	}
+
+	err := opts.newHostIds()
+	if err != nil {
+		return err
+	}
+
+	// shutdown cluster
+	err = opts.shutdownCluster()
+	if err != nil {
+		return err
+	}
+
+	// Remove cluster from automation
+	err = opts.removeClusterFromAutomation()
+	if err != nil {
+		return err
+	}
+
+	// Stop monitoring
+	err = opts.stopMonitoring()
+	if err != nil {
+		return err
+	}
+
+	fmt.Print("Cluster deleted\n")
+	return nil
+}
+
+func (opts *DeleteOpts) newHostIds() error {
 	current, err := opts.store.GetAutomationConfig(opts.ConfigProjectID())
+	if err != nil {
+		return err
+	}
+	r := convert.FromAutomationConfig(current)
+	hostname := make(map[string][]int32)
+
+	for _, rs := range r {
+		if rs.Name == opts.Entry {
+			for _, s := range rs.ProcessConfigs {
+				hostname[s.Hostname] = append(hostname[s.Hostname], int32(s.Port))
+			}
+		}
+	}
```

Comment:
[nit] this is going to be quite slow, convert to the cli presentation just to group per replica set, to then find the replicaset you want, processes have [cluster field](https://github.com/mongodb/go-client-mongodb-ops-manager/blob/bfe7ba34b319e2a9fb17a0d60f59f5b3c5cbc07d/opsmngr/automation_config.go#L330) which should match the name of the cluster you're deleting, you can then just search over all the processes by this field